### PR TITLE
Bttribev2 10270 range slider

### PR DESCRIPTION
**Pull request by AdimAnyvision**
## What I did
I did a range slider component. The slider's elements are range input, tooltip to show the current value / current hovered value of the slider, and min/ max labels.
The slider get min, max, value, onChange, step, disabled, width and isToggleTooltip as props, if not given it gets the default.
Other props can be added like className, style.

## How to test
Run storybook and checkout the examples, the component's functions and helpers are documented so it will be easier to understand the functions and calculations

### Checklist
- [ ] Does this need a new/update storybook example?
- [ ] Does this need new/update tests?

- [ ] run eslint
- [ ] run test
- [ ] run build

If your answer is yes to any of these, please make sure to include it in your PR.

> NOTE: Please submit all PRs to the `development` branch.

### CODE REVIEW

File: src/components/RangeSlider/RangeSlider.js [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+  /** Value of the slider */
+  value: propTypes.oneOfType([propTypes.string, propTypes.number]),
+  /** Callback when the component's state is changed. */
+  onChange: propTypes.func,
+  /** The step by which the value is incremented / decremented. */
+  step: propTypes.number,
+  /** Determines the disabled mode of the RangeSlider, if true - disabled. */
+  disabled: propTypes.bool,
+  /** The width of the range slider. Default is 480 = 480px. */
+  sliderWidth: propTypes.number,
```
Comment (ozAnyvision) [inline src/components/RangeSlider/RangeSlider.js:240]: Didn't we discuss about the width and said it's not needed?

**Review by ozAnyvision — CHANGES_REQUESTED** [review]
@IrenaSAnyVision can you please review the PR as well?

File: src/components/RangeSlider/RangeSlider.js (hunk 1 above)
Comment (AdimAnyvision) [reply src/components/RangeSlider/RangeSlider.js:240]: It iinitially was with fixed widths (144, 240, 360, 480, 640, 960) and as far as I remember we said we will allow the width to be determined manually.
With the disabled, I don't remember. I removed the isDarkTheme prop. 
Do you want me to remove both of the props ?

File: src/components/RangeSlider/RangeSlider.js [hunk 2]
```diff
@@ -0,0 +1,246 @@
+import React, { useState, useRef, useEffect, useCallback, useMemo } from 'react'
+import styles from './RangeSlider.module.scss'
+import propTypes from 'prop-types'
+import classNames from 'classnames'
```
[the remark this answers is not in the document]
Comment (AdimAnyvision) [reply src/components/RangeSlider/RangeSlider.js:4]: Ok, I will change

File: src/components/RangeSlider/RangeSlider.js [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+  const renderTooltip = () => {
+    const tooltipEl = (
+      <div ref={tooltipRef} className={styles.tooltip}>
+        {hoverValue ?? value}
+      </div>
+    )
+
+    if (isToggleTooltip) return showTooltip && tooltipEl
+    return tooltipEl
+  }
```
Comment (IrenaSAnyVision) [inline src/components/RangeSlider/RangeSlider.js:189]:
```suggestion
  const renderTooltip = () => (!isToggleTooltip || showTooltip) ? (
    <div ref={tooltipRef} className={styles.tooltip}>
      {hoverValue ?? value}
    </div>
  ) : null
  }
```
I think it has the same result - check it and choose what you prefer

**Review by IrenaSAnyVision — CHANGES_REQUESTED** [review]

File: src/components/RangeSlider/RangeSlider.js [hunk 4]
```diff
@@ -0,0 +1,246 @@
+import React, { useState, useRef, useEffect, useCallback, useMemo } from 'react'
```
Comment (IrenaSAnyVision) [inline src/components/RangeSlider/RangeSlider.js:1]: please add a line to the CHANGELOG.md file about your component :)

File: src/components/RangeSlider/RangeSlider.js [hunk 5]
```diff
@@ -165,28 +160,41 @@ const RangeSlider = ({
...
 
+  // KeyDown is called while user presses the arrows and KeyUp is called when the press
+  // is done, so the input value will be updated and won't be one step ahead.
+  const handleKeyPress = e => {
+    if (e.keyCode === 37 || e.keyCode === 39) {
+      posTooltipToValue()
+    }
+  }
+
+  const renderMeasureUnitText = () => measureUnitText || null
```
Comment (IrenaSAnyVision) [inline src/components/RangeSlider/RangeSlider.js:171]: Is the null necessary?

File: src/components/RangeSlider/RangeSlider.js [hunk 6]
```diff
@@ -165,28 +160,41 @@ const RangeSlider = ({
     setHoverValue(value)
   }
 
+  // KeyDown is called while user presses the arrows and KeyUp is called when the press
+  // is done, so the input value will be updated and won't be one step ahead.
+  const handleKeyPress = e => {
+    if (e.keyCode === 37 || e.keyCode === 39) {
```
Comment (IrenaSAnyVision) [inline src/components/RangeSlider/RangeSlider.js:166]: there is a keymap in: utils/enums/keymap.js
please use it instead of writing 37 or 39 in the code

**Review by IrenaSAnyVision — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -64,6 +64,7 @@
 - New Text Field.
 - New Wizard Dialog.
 - Added default behavior that doesn't require isOpen in Menu
+- New Range Slider.
```
Comment (IrenaSAnyVision) [inline CHANGELOG.md:67]: Almost XD
The file is separated by the package versions..
You added your "text" to version 0.1.3 (that was released on September last year)
You should add it to "Added" part of "Unreleased" version (line 39)

And please write it as on line 38..

File: src/components/RangeSlider/RangeSlider.js [hunk 8]
```diff
@@ -163,7 +166,7 @@ const RangeSlider = ({
   // KeyDown is called while user presses the arrows and KeyUp is called when the press
   // is done, so the input value will be updated and won't be one step ahead.
   const handleKeyPress = e => {
-    if (e.keyCode === 37 || e.keyCode === 39) {
+    if (e.keyCode === keymap.ARROW_LEFT || e.keyCode === keymap.ARROW_RIGHT) {
```
Comment (IrenaSAnyVision) [inline src/components/RangeSlider/RangeSlider.js:169]:
```suggestion
    if ([keymap.ARROW_LEFT, keymap.ARROW_RIGHT].includes(e.keyCode)) {
```
Another way to write it (choose what you prefer - I don't mind)

File: src/components/RangeSlider/RangeSlider.js [hunk 9]
```diff
@@ -1,6 +1,7 @@
 import React, { useState, useRef, useEffect, useCallback, useMemo } from 'react'
 import propTypes from 'prop-types'
 import classNames from 'classnames'
+import keymap from '../../utils/enums/keymap'
```
Comment (IrenaSAnyVision) [inline src/components/RangeSlider/RangeSlider.js:4]: I still do not see the export of the component......

File: src/components/RangeSlider/RangeSlider.js (hunk 8 above)
Comment (IrenaSAnyVision) [reply src/components/RangeSlider/RangeSlider.js:169]: please try to press the keys UP & DOWN or HOME & END or PG UP & PG DN
For my experiments they all moves the "circle" but it does not affect the "tooltip" above

**Review by IrenaSAnyVision — CHANGES_REQUESTED** [review]

**Review by IrenaSAnyVision — APPROVED** [review]

**Review by IrenaSAnyVision — APPROVED** [review]

**Review by ozAnyvision — APPROVED** [review]

**Pull request merged**